### PR TITLE
Bridge: remove unwrap, expect and some warnings

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

Remove unwrap/expect that can panic the relayer.

# Changelog
Remove unwrap and expect that can panic the relayer.
Remove some warning
Change From conversion that can fail to TryFrom.

# Testing

# Outstanding issues

### CODE REVIEW

**Review by andygolay — COMMENTED** [review]
When I run a local node with 

```
CELESTIA_LOG_LEVEL=FATAL CARGO_PROFILE=release CARGO_PROFILE_FLAGS=--release  nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash  -c "just bridge native build.setup.eth-local.celestia-local.bridge --keep-tui"
```

 and run the test
 
```
DOT_MOVEMENT_PATH="$(pwd)/.movement" cargo test --test bridge_e2e_test_framework test_bridge_transfer_eth_movement_happy_path -- --nocapture --test-threads=1
```

I get 

```
2024-11-04T16:01:04.650271Z  INFO bridge_e2e_test_framework: Call initiate_transfer on Eth
base_call_builder: CallBuilder { request: TransactionRequest { from: Some(0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc), to: Some(Call(0x8a791620dd6260079bf849dc5567adc3f2fdc318)), gas_price: None, max_fee_per_gas: None, max_priority_fee_per_gas: None, max_fee_per_blob_gas: None, gas: None, value: Some(1), input: TransactionInput { input: Some(0x1a8849a40000000000000000000000000000000000000000000000000000000000000000e06c5f4faa4b2a4b3ef18c495c9232c8f25f294f1906abd844547a154bac97434dc9b1ea97469e2b1f12d5a7db9ab4b56566a74aa6761a604e75eaa20796f153), data: None }, nonce: None, chain_id: None, access_list: None, transaction_type: None, blob_versioned_hashes: None, sidecar: None }, block: Number(Latest), state: None, decoder: "bridge_service::chains::ethereum::types::AtomicBridgeInitiator::initiateBridgeTransferCall" }
Sending transaction with gas limit: 10000000000000000
thread 'test_bridge_transfer_eth_movement_happy_path' panicked at protocol-units/bridge/integration-tests/tests/bridge_e2e_test_framework.rs:107:6:
Failed to initiate bridge transfer: Generic error: Failed to send transaction: server returned an error response: error code 3: execution reverted
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
FAILED
```

Are there other tests you recommend to run for this, that you could add to the PR description?

**Comment by musitdev** [comment]
Yes, the e2e test is broken since the eth move token change PR. There's no test besides starting the bridge to see if it starts.

**Review by andygolay — APPROVED** [review]
LGTM

File: protocol-units/bridge/service/src/chains/ethereum/types.rs [hunk 1]
```diff
@@ -147,89 +148,92 @@ impl std::ops::Deref for EthAddress {
...
+// 			AlloyParam::Amount => Param {
+// 				name: "amount".to_string(),
+// 				ty: "uint256".to_string(),
+// 				components: vec![],
+// 				internal_type: None,
+// 			},
+// 		}
+// 	}
+// }
+
```
Comment (0xmovses) [inline protocol-units/bridge/service/src/chains/ethereum/types.rs:236]: Let's delete this commented out code, it's no longer needed at all.

**Review by 0xmovses — CHANGES_REQUESTED** [review]
Looks good, one request.

**Review by 0xmovses — APPROVED** [review]

**Pull request merged**